### ISSUE TITLE
D3 project Bar Chart: Test not aligned with description.

### ISSUE THREAD

**Issue by FrankGrimm** [freeCodeCamp/freeCodeCamp#39299]
**Problem description:**

The test for the Bar Chart project in the Data Visualization category is not aligned with the corresponding user stories.

"**User Story #10**: The data-date attribute and its corresponding bar element should align with the corresponding value on the x-axis." does not specify that the x-axis may only contain years instead of full dates.

[Description](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/04-data-visualization/data-visualization-projects/visualize-data-with-a-bar-chart.english.md) and [Corresponding test](https://github.com/freeCodeCamp/testable-projects-fcc/blob/master/src/project-tests/bar-chart-tests.js#L176-L206)

That test passes `dataType = 'year'` into [`areShapesAlignedWithTicks`](https://github.com/freeCodeCamp/testable-projects-fcc/blob/master/src/utils/alignment-D3.js#L33) which, in [`getTickValue`](https://github.com/freeCodeCamp/testable-projects-fcc/blob/master/src/utils/alignment-D3-support.js#L33) only returns the innerHTML of the tick, while parsing out the year of the date in the tooltip ([`getShapeValue`](https://github.com/freeCodeCamp/testable-projects-fcc/blob/master/src/utils/alignment-D3-support.js#L52)).

Expected behaviour would be that this is either documented in the requirements (instead of only the working example) or that the test be made more lenient.

**Comment by ryanwinchester** [comment]
It's very confusing

**Comment by dylanesque** [comment]
What needs to happen for some forward motion on this? Barring that, what would the workaround be for this that doesn't break other tests? I'd be more than happy to help if there's something I could do!

**Comment by raisedadead** [comment]
@freecodecamp/moderators can you please help with this?

**Comment by moT01** [comment]
As stated in the issue - seems like there's two options:
- Adjust the test to allow any type of date - (fixed in testable-projects repo)
- Adjust the user story to say that it has to be the year - (need a PR to testable-projects and main repos)

The first would probably be the better solution. But, either way works for me. Any interest in working on this @dylanesque?

**Comment by dylanesque** [comment]
> As stated in the issue - seems like there's two options:
> 
> * Adjust the test to allow any type of date - (fixed in testable-projects repo)
> * Adjust the user story to say that it has to be the year - (need a PR to testable-projects and main repos)
> 
> The first would probably be the better solution. But, either way works for me. Any interest in working on this @dylanesque?

Yeah, I'll give this a go with a preference for the first solution.

**Comment by dylanesque** [comment]
Before I get started on this PR, I need some additional clarification: while working on the issue that led me to find this issue, I ran into something odd based on a forum suggestion to fix this test in my code. The repo is at (https://github.com/dylanesque/BarChart), the short version is that this test is failing for me because it seems to be reading the axis label as a tick, and throwing the 'innerHTML of null' error. Is this a problem with my implementation, or is this an additional bug with the test?

**Comment by moT01** [comment]
I looked into this a little deeper. I believe D3 automatically gives the `tick` class to the elements when generating an axis. You have added the `tick` class to your [axis label](https://github.com/dylanesque/BarChart/blob/master/chart.js#L114). So it's trying to test it as if it's one of the ticks. Changing the class there to something else will pass the tests. I don't think that is the same issue as described here, and I'm not sure if it's worth trying to account for that edge case. But I wouldn't object to possible solutions.

I still think the solution for this issue is to do one of the two things I had previously suggested.

**Comment by dylanesque** [comment]
> I looked into this a little deeper. I believe D3 automatically gives the `tick` class to the elements when generating an axis. You have added the `tick` class to [your axis label](https://github.com/dylanesque/BarChart/blob/master/chart.js#L114). So it's trying to test it as if it's one of the ticks. Changing the class there to something else will pass the tests. I don't think that is the same issue as described here, and I'm not sure if it's worth trying to account for that edge case. But I wouldn't object to possible solutions.
> 
> I still think the solution for this issue is to do one of the two things I had previously suggested.

Thanks for taking a look, you're right, manually adding that class is what was causing the problem!

**Comment by moT01** [comment]
Do you have an example project that was having this issue @FrankGrimm?

**Comment by FrankGrimm** [comment]
> Do you have an example project that was having this issue @FrankGrimm?

No, sorry, I was just helping a friend that took the course back then and thought I'd file an issue on it.
